src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -314,6 +336,37 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         End Function
 
         <Extension()>
+        Friend Function AreSameCustomModifiers([mod] As ImmutableArray(Of CustomModifier), otherMod As ImmutableArray(Of CustomModifier)) As Boolean
+            Dim count As Integer = [mod].Length
+
+            If (count <> otherMod.Length) Then
+                Return False
+            End If
+
+            Return [mod].SequenceEqual(otherMod)
+        End Function
+
+        Private Function HasSameTupleNames(t1 As TypeSymbol, t2 As TypeSymbol) As Boolean
+            If Not t1.IsTupleType AndAlso
+                Not t2.IsTupleType Then
+                Return True
+            End If
+
+            Dim t1Names = t1.TupleElementNames
+            Dim t2Names = t2.TupleElementNames
+
+            If t1Names = Nothing AndAlso t2Names = Nothing Then
```

Comment:
>t1Names = Nothing [](start = 15, length = 17)

It looks like we are comparing an ImmutableArray, which is a structure, with Nothing. Do you know what code does this actually generate? In general, we are avoiding comparisons like this against structures. Consider using IsDefault property instead.